merkledag.go

```diff
@@ -7,7 +7,37 @@ import (
 	cid "github.com/ipfs/go-cid"
 )
 
-var ErrNotFound = fmt.Errorf("merkledag: not found")
+// ErrNotFound is used to signal when a Node could not be found. The specific
+// meaning will depend on the DAGService implementation, which may be trying
+// to read nodes locally but also, trying to find them remotely.
+var ErrNotFound = ErrNotFoundCid{}
```

Comment:
Returning `ErrNotFoundCid{Cid}` is already going to break anything comparing against `ErrNotFound`. Given that, I'd rather _explicitly_ break it and replace `ErrNotFound` with `type ErrNotFound struct { Cid }`.

Reply:
Ok, but no-one returns ErrNotFoundCid (and we can grep-fix all of our things accordingly). So for everyone else things would still work. But yeah, let's play safe and break.